sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingDeduplicationSuite.scala

```diff
@@ -484,6 +486,52 @@ class StreamingDeduplicationSuite extends StateStoreMetricsTest {
       CheckLastBatch(("c", 9, "c"))
     )
   }
+
+  test("collation aware deduplication") {
+    val inputData = MemoryStream[(String, Int)]
+    val result = inputData.toDF()
+      .select(col("_1")
+        .try_cast(StringType("UNICODE")).as("str"),
+        col("_2").as("int"))
+      .dropDuplicates("str")
+
+    testStream(result, Append)(
+      AddData(inputData, "a" -> 1),
+      CheckLastBatch("a" -> 1),
+      assertNumStateRows(total = 1, updated = 1, droppedByWatermark = 0),
+      AddData(inputData, "a" -> 2), // Dropped
+      CheckLastBatch(),
+      assertNumStateRows(total = 1, updated = 0, droppedByWatermark = 0),
+      // scalastyle:off
+      AddData(inputData, "ä" -> 1),
+      CheckLastBatch("ä" -> 1),
+      // scalastyle:on
+      assertNumStateRows(total = 2, updated = 1, droppedByWatermark = 0)
+    )
+  }
+
+  test("non-binary collation aware deduplication not supported") {
+    val inputData = MemoryStream[(String)]
+    val result = inputData.toDF()
+      .select(col("value")
+        .try_cast(StringType("UTF8_BINARY_LCASE")).as("str"))
```

Comment:
I'm a bit confused by this - is the test name flipped or is UTF8_BINARY_LCASE considered a non-binary collation?

Reply:
AFAIK, LCASE means comparing as lowercase, so yes it's bound to non-binary equality.

Reply:
That's right, UTF8_BINARY_LCASE is a non-binary collation.

In other words, "AAA" and "aaa" are considered equal, even though binary representations are clearly different.